features/machine/machine_healthcheck.feature

```diff
@@ -166,5 +166,44 @@ Feature: MachineHealthCheck Test Scenarios
      | Ensuring a requeue happens in 15m0  |
     """
    Then the machine should be remediated
+ 
+  # @author miyadav@redhat.com
+  # @case_id OCP-29857
+  @admin
+  @destructive
+  Scenario: [MHC] MaxUnhealthy should not allow malformed values
+    Given I have an IPI deployment
+    And I switch to cluster admin pseudo user
+    Then I use the "openshift-machine-api" project
+    And admin ensures machine number is restored after scenario
+
+    Given I clone a machineset and name it "machineset-clone-29857"
+
+    # Create MHC with malformed unhealthy nodes value
+    When I run oc create over "<%= BushSlicer::HOME %>/testdata/cloud/mhc/mhc_malformed.yaml" replacing paths:
+     | n                                                                                  | openshift-machine-api       |
+     | ["metadata"]["name"]                                                               | mhc-<%= machine_set.name %> |
+     | ["spec"]["selector"]["matchLabels"]["machine.openshift.io/cluster-api-cluster"]    | <%= machine_set.cluster %>  |
+     | ["spec"]["selector"]["matchLabels"]["machine.openshift.io/cluster-api-machineset"] | <%= machine_set.name %>     |
+    Then the step should succeed
+    And I ensure "mhc-<%= machine_set.name %>" machinehealthcheck is deleted after scenario
+ 
+    Given I create the 'Ready' unhealthyCondition
```

Comment:
No need to create the unhealthycondition as well. You can just run the next step to see the log